spec/system/purchase_system_spec.rb

```diff
@@ -301,14 +301,50 @@
       sign_in organization_admin
     end
 
-    it "allows deletion of a purchase" do
-      visit "#{subject}/#{purchase.id}"
-      expect(page).to have_link("Delete")
-      accept_confirm do
-        click_on "Delete"
+    context "When the purchase remains in storage location" do
+      it "allows deletion of a purchase" do
+        visit "#{subject}/#{purchase.id}"
+        expect(page).to have_link("Delete")
+        accept_confirm do
+          click_on "Delete"
+        end
+        expect(page).to have_content "Purchase #{purchase.id} has been removed!"
+        expect(page).to have_content "0 (Total)"
+      end
+    end
+
+    context "When the purchase has been distributed" do
```

Comment:
Really all we're testing here is the ability to catch errors. So you can ensure that the delete service throws an error:

```ruby
allow(PurchaseDestroyService).to receive(:call).and_raise('an error')
```

and you don't need all the setup here.

Reply:
Oh, that's really easy. I will change it. Thanks for your review.